src/v/kafka/data/cloud_topic_partition_reader.cc

```diff
@@ -74,11 +76,27 @@ cloud_topic_partition_reader_impl::cloud_topic_partition_reader_impl(
   ss::shared_ptr<experimental::cloud_topics::data_plane_api> ct_api)
   : _config(cfg)
   , _underlying(std::move(underlying))
-  , _ct_api(std::move(ct_api)) {}
+  , _ct_api(std::move(ct_api)) {
+    if (_config.max_bytes == 0) {
+        _config.max_bytes = L0_default_bytes_per_fetch;
+    }
+    _config.max_bytes = std::clamp(
+      _config.max_bytes, 0UL, L0_max_bytes_per_fetch);
+}
```

Comment:
what's up with this? It's not mentioned why this happened in the comit

Reply:
that's just a sanity check for the max_bytes, I'll add a comment